src/components/Graph.react.js

```diff
@@ -55,22 +83,23 @@ class PlotlyGraph extends Component {
         ) {
             extendData.push(nextProps.extendData);
         } else {
-            extendData = EMPTY_EXTEND_DATA;
+            extendData = EMPTY_DATA;
         }
 
-        if (extendData !== EMPTY_EXTEND_DATA) {
+        if (extendData !== EMPTY_DATA) {
             this.setState({
                 extendData,
             });
         }
     }
 
-    clearExtendData() {
-        this.setState(({extendData}) => {
+    clearState(dataKey) {
+        this.setState(props => {
+            var data = props[dataKey];
             const res =
-                extendData && extendData.length
+                data && data.length
                     ? {
-                          extendData: EMPTY_EXTEND_DATA,
+                        data: EMPTY_DATA,
```

Comment:
```suggestion
                        [dataKey]: EMPTY_DATA,
```

Not sure how that would cause the prepend tests to fail while the extend tests still succeed... there may be something else funny going on. Aside from the fact that it broke the tests, yes this refactoring is exactly what I had in mind 😅 

Might be easier to debug those locally with `pytest -k test_graph_prepend_trace` or, if you want to play with the test app before the automation commands run, `pytest -k test_graph_prepend_trace --pause`

(also the lint test failed, but that can be fixed with with `npm run format`)

Reply:
Not super familiar with JavaScript yet. What is that `[..]` syntax you are proposing?

Reply:
Is there a guide on how to get the whole Chrome thing working for tests? I am on Windows with WSL and have not got it working yet.

Reply:
When running the prepend and extend examples isolated on my machine (I bascially extracted the test code into a separate dash app .py file) then it works for both prepend and extend. Weird.

Reply:
Finally got selenium working after upgrading to WSL 2. Will look into this in the coming days.

Reply:
Oh it took me so long to understand your suggestion... Total Javascript/React rooky here